src/api.js

```diff
@@ -1,58 +1,64 @@
 import http from './http'
 
-const OPTIONS = {
-  method: 'GET',
-  body: null,
+/**
+ * 
+ * A lightweight wrapper around the {@link http} module.
+ * Provides functions to make API requests with specified HTTP methods.
+ *
+ * The functions are as follows:
+ * - `get(url, options)` sends a `'GET'` request
+ * - `patch(url, body, options)` sends a `'PATCH'` request
+ * - `post(url, body, options)` sends a `'POST'` request
+ * - `put(url, body, options)` sends a `'PUT'` request
+ * - `destroy(url, body, options)` sends a `'DELETE'` request
+ * - `call(url, method, body, options)` sends a request with specified method
+ *
+ * Each function can be passed an `options` object, which will eventually be forwarded
+ * to the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).
+ * 
+ * Each function returns a promise, which will either resolve with a response object
+ * or reject with an {@link HTTPError}.
+ *
+ * @name api
+ * @type Object
+ * 
+ * @example
+ * 
+ * function getUsers () {
+ *   return api.get('/users', { credentials: 'include' })
+ * }
+ * 
+ * getUsers()
+ *    .then(res => console.log('The users are', res))
+ *    .catch(err => console.log('An error occurred!', err))
+ */
+
+function requestWithMethod (url, body, opts, method) {
+  return http(url, { ...opts, body, method })
 }
 
 export function get(url, opts = {}) {
-  return http(url, {
-    ...OPTIONS,
-    ...opts,
-  })
+  const body = null
```

Comment:
its possible this could cause a problem, I think changing it from `null` to `undefined` might be safer OR change the `omitUndefined` calls to also omit `null`. 

Reason: `undefined` values are explicitly omitted from calls to `fetch` because for some reason `fetch` expects a value if the key exists. maybe test this out to make sure nothing breaks.